src/main/kotlin/com/codingchallenge/transactionanalyser/TransactionLoader.kt

```diff
@@ -0,0 +1,35 @@
+package com.codingchallenge.transactionanalyser
+
+import com.codingchallenge.transactionanalyser.domain.Transaction
+import com.codingchallenge.transactionanalyser.domain.TransactionBuilder
+import java.io.BufferedReader
+import java.io.InputStream
+import java.io.InputStreamReader
+import java.util.stream.Collectors
+
+class TransactionLoader(private val input: InputStream) {
+    fun loadTransactionsFromCsv(): List<Transaction>{
+        val transactions: List<Transaction>
+        BufferedReader(InputStreamReader(input)).use {
+            transactions = it.lines()
+                .skip(1)
+                .map(this::buildTransactionFromCsvLine)
+                .collect(Collectors.toList())
+        }
+        return transactions
+    }
+
+    private fun buildTransactionFromCsvLine(line: String): Transaction{
+        val values = line.split(",").map { it.trim() }
+        var transactionBuilder = TransactionBuilder(values[0])
+            .withFromAccountId(values[1])
+            .withToAccountId(values[2])
+            .withCreatedAt(values[3])
+            .withAmount(values[4])
+            .withType(values[5])
+        if (values.size == 7){
```

Comment:
you are checking the size of the number of columns to determine if the transaction is a reversal. 
What if I added more columns with different data? 
Bring your code design closer to the data itself so you don't have to introduce this "7 fields equals a reversal" rule that you have essentially created here. What I mean by that is you introduced a concept, a rule, something that is unique to this codebase but not if the data structure changes. The CSV file **has** a field with the label `REVERSAL`. That is the way you know it's a reversal from the data itself. Perhaps we could change our code to use _that_ as the only rule that it is a reversal, not if it has a certain number of columns.

Be adaptable to change and think about the rules you are introducing yourself that is different from the 'rules of the data' in this case.

Reply:
Yeah, I can see that it's a bad idea.